app/models/user_project_role.rb

```diff
@@ -0,0 +1,15 @@
+class UserProjectRole < ActiveRecord::Base
```

Comment:
@henders : I have a few questions in here:

First one: according to the documentation, we should use `has_many` when we need an extra attribute:
> You should use has_many :through if you need validations, callbacks, or extra attributes on the join model.

Beside project and user, this user_project_role table contains a role_id column. Doesn't it prevent us from using the `has_and_belongs_to_many` relationship?

Second: after going through the documentation, I didn't fully understand your example, the part where you added `:ProjectRole` on the `has_and_belongs_to_many`. I would have understood something like `has_and_belongs_to_many :projects`, for User, and `has_and_belongs_to_many :users`, for Project. I also didn't find anything that could explain why `:ProjectRole`. Was it a typo? Just asking before I change the whole thing. :)

Finally: `ProjectRole` is not an `ActiveRecord::Base` descendant, instead, it inherits from `ActiveHash::Base`. I know ActiveHash allows us to treat a hash as a model-like entity, but doesn't it stop there? 

So, in the end, let me know if I'm doing the right thing or not. :)

One thing I noticed is that I was missing the `has_many :projects, through: :user_project_roles`, and the corresponding one for the the Project model, to get the users. It may come in handy.